source/includes/steps-fle-configure-the-mongodb-client.yaml

```diff
@@ -137,6 +163,25 @@ content: |
                const extraOptions = {
                  mongocryptdBypassSpawn: true,
                }
+     .. tab::
+        :tabid: python
+
+        .. code-block:: python
+      
+           mongocryptd_spawn_path = "/usr/local/bin/mongocryptd"
+
+        .. admonition:: Encryption Binary Daemon
+           :class: note
+
+           If the ``mongocryptd`` daemon is already running, you can
+           configure the client to skip starting it by passing the
+           following option:
+
+           .. code-block:: python
+              :emphasize-lines: 1
+
+               mongocryptd_bypass_spawn = True
```

Comment:
The other examples show where this variable gets passed in to construct `extraOptions` that gets passed to the MongoClientSettings. Rather than creating a variable, it could be better to show it directly as an optional parameter of `AutoEncryptionOpts`. The reason for this is that the code snippet that constructs the `AutoEncryptionOpts` in the next step uses this as a variable, but the snippet will likely **not work** if the value of the variable is not set. This would be common to any user that does not explicitly define the variable.

Reply:
For example, something like this may suffice (as well as a short explanation that the default value for the optional parameter is False):
```
fle_opts = AutoEncryptionOpts(
    ...
    mongocryptd_bypass_spawn = True
)
```

Reply:
Another option is to mention where you would pass the following option around line 178 and then omit it from the next step.

Reply:
I'd actually prefer to keep it the way that it is. The structure that the Node/Java examples use is:

In step four, Specify the Location of the Encryption Binary: they define extraOptions with a mongocryptdBypassSpawn variable
In step five, Create the MongoClient: they use the extraOptions defined in the last step (but do not redefine it) and pass it into the auto encryption settings.
The python driver unfortunately doesn't use the object/dict "extraOptions". Instead we pass it directly to the auto encryption settings/options. Thus I am following the same general pattern for python:

In step four, Specify the Location of the Encryption Binary: python snippet defines a mongocryptd_bypass_spawn variable
In step five, Create the MongoClient: python snippet uses the mongocryptd_bypass_spawn defined in the last step (but does not redefine it) and passes it into the auto encryption settings.
I think if it is safe to assume that the readers will either copy and paste the extraOptions from the last section or not (and receive an error), then it is also same to assume the same for mongocryptd_bypass_spawn

Reply:
I think I see the misunderstanding. This comment is going to be a little lengthy, so feel free to ask me in person to go through it.

To summarize my request:
-Since setting mongocryptd_bypass_spawn is mentioned as optional, we should not assume that it has been defined. It has a default value, but if we pass `mongocryptd_bypass_spawn = mongocryptd_bypass_spawn` and the variable has not been defined, it will error.

To summarize what I read from your response:
1) Python (or pymongo) does not have the same call syntax as Java to allow us to make it optionally included.
2) The reader will just copy and paste extraOptions from the prior section (assuming you meant from the same step

For 2, are you referring to "B. Create a Data Encryption Key Step 4"? This is two sections prior to "D. Create the MongoDB Client". I think for effective and maintainable guides, we should try to keep sections self-enclosed.

For 1, I was not concerned there is a feature of python2/3 called "unpacking" that allows you to create something very similar to the Java example.
Here's a link to the python doc: https://docs.python.org/3/tutorial/controlflow.html#tut-unpacking-arguments
Here's a sample:

```
def my_func(a, b = True, c = "mongodb://localhost:27017"):
   print("a:", a)
   print("b:", b)
   print("c:", c)

>>> opts = { "hello": "false", "c": "mongodb://localhost:27018" }
>>> my_func("foobar", **opts)
```

From looking at the Node example, I'd like to request a change to that snippet too. When showing`mongocryptd_bypass_spawn`, instead of showing it as the only parameter of `extraOptions` (const), append it to extraOptions (var) in the admonition.

Reply:
Fixed, switched snippets roughly to:

```
           extra_options = {
              'mongocryptd_spawn_path': '/usr/local/bin/mongocryptd'
           }
           extra_options['mongocryptd_bypass_spawn'] = True
           fle_opts = AutoEncryptionOpts( 
              kms_providers, 
              key_vault_namespace,
              schema_map = patient_schema,
              **extra_options
           )
           client = MongoClient(connection_string, auto_encryption_opts = fle_opts)
```